flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -377,6 +378,28 @@ public ComponentMainThreadExecutor getJobMasterMainThreadExecutor() {
         return Optional.ofNullable(checkpointStorageName);
     }
 
+    @Override
+    public ExecutionVertex getExecutionVertex(final ExecutionVertexID executionVertexId) {
+        final Map<JobVertexID, ExecutionJobVertex> executionJobVertices = getAllVertices();
+        Preconditions.checkArgument(
+                executionJobVertices.containsKey(executionVertexId.getJobVertexId()),
+                "The passed executionVertexId {} is invalid: There is no ExecutionJobVertex with the associated JobVertexId {}.",
+                executionVertexId,
+                executionVertexId.getJobVertexId());
+
+        final ExecutionJobVertex executionJobVertex =
+                executionJobVertices.get(executionVertexId.getJobVertexId());
```

Comment:
Do we really need to add another method to the EG for something that is already well supported with existing APIs?
```
AccessExecutionVertex vertex = accessExecutionGraph.getJobVertex(executionVertexId.getJobVertexId())[executionVertexId.getSubtaskIndex()];
```